src/wallet/coinselection.cpp

```diff
@@ -164,137 +164,29 @@ bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_va
     return true;
 }
 
-static void ApproximateBestSubset(const std::vector<CInputCoin>& vValue, const CAmount& nTotalLower, const CAmount& nTargetValue,
-                                  std::vector<char>& vfBest, CAmount& nBest, int iterations = 1000)
+bool SingleRandomDraw(const CAmount& target_value, std::vector<CInputCoin>& utxo_pool, std::set<CInputCoin>& out_set, CAmount& value_ret, CAmount non_input_fees)
```

Comment:
Theoretically, SRD could also produce a result without a change. You could fold the `non_input_fees` into the `target_value`, and if you don't find a perfect match, select more until you have minChange + fee for one output more than target.

Reply:
I don't think that this distinction matters. If SRD produces a result without change, the rest of the code will treat it as if BnB had been used, there is no difference for change or no change. I don't think it is necessary to target a min change or include the change output's fees in the target.